### DISCUSSION TITLE
Dynamic Import Snippet for React with Typescript

### DISCUSSION THREAD

**Discussion by charlieforward9** [vikejs/vike#693] [Discussion] [answered]
I am having a lot of trouble implementing a client-only component. [Dynamic Imports](https://github.com/brillout/vite-plugin-ssr/blob/main/docs/pages/dynamic-import.page.server.mdx) most closely fit my use case.

I am still building experience in JS/TS/React, so it may be an experience thing, but the snippet provided throws a bug when I paste it, and I can't find my way around it.

` setComponent(() => React.lazy(() => import("../../components/Map")));` in the snippet throws
> **Type 'LazyExoticComponent<() => Element>' is not assignable to type 'Element | (() => Element)'.**

where the path exports:

``` typescript
import React from "react";
import { MapContainer, TileLayer, Marker, Popup } from "react-leaflet";

export default function Map() {
  return (
    <div className="h-full w-full flex">
      <MapContainer center={[51.505, -0.09]} zoom={13} scrollWheelZoom={false}>
        <TileLayer
          attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />
        <Marker position={[51.505, -0.09]}>
          <Popup>
            A pretty CSS3 popup. <br /> Easily customizable.
          </Popup>
        </Marker>
      </MapContainer>
    </div>
  );
}


```

**Comment by charlieforward9** [comment]
I renamed `Map` component in [this snippet](https://github.com/brillout/vite-plugin-ssr/blob/main/docs/pages/dynamic-import.page.server.mdx) to `LazyMap`

**Comment by charlieforward9** [comment, marked as answer]
Seems like this is a known problem [also mentioned here](https://youtrack.jetbrains.com/issue/WEB-45615). It not due to the plugin, but I do think it is something worth mentioning in [the docs](https://github.com/brillout/vite-plugin-ssr/blob/main/docs/pages/dynamic-import.page.server.mdx) for Typescript-based implementations.

**Reply by brillout** [reply]
Did you end up using `@ts-ignore` or did you find a solution?

> worth mentioning in the docs 

Agreed. PR welcome.

**Reply by charlieforward9** [reply]
I just ran the code with the error, and it works fine. You just taught me about `@ts-ignore`. Thank you! That's what I will go with.
